### PR TITLE
egnkey cli functionality

### PR DESCRIPTION
**Pull request by ricomateo**
**Description**
This PR adds the following CLI commands
* `egnaddrs`
* `egnkey`
    * `generate`
    * `convert`
    * `derive-operator-id`

### CODE REVIEW

**Comment by supernovahs** [comment]
this needs some changes after #69 is merged

File: crates/eigen-cli/src/args.rs
Comment (supernovahs) [inline crates/eigen-cli/src/args.rs:95]: can we name this as something ConvertECDSA? since i am working on a keystore integartion for bls also, it will be confusing

**Review by supernovahs — CHANGES_REQUESTED** [review]
naming for ecdsa keystore

File: crates/eigen-cli/src/args.rs
Comment (ricomateo) [reply crates/eigen-cli/src/args.rs]: Done!

File: crates/eigen-cli/src/generate.rs
Comment (dralves) [inline crates/eigen-cli/src/generate.rs:15]: why so short?

File: crates/eigen-cli/src/lib.rs
Comment (dralves) [inline crates/eigen-cli/src/lib.rs:16]: separate the module declarations form imports

File: crates/eigen-cli/src/lib.rs
Comment (supernovahs) [inline crates/eigen-cli/src/lib.rs:84]: print remove  pending

File: crates/eigen-cli/src/lib.rs
Comment (supernovahs) [inline crates/eigen-cli/src/lib.rs:112]: same

File: crates/eigen-cli/src/generate.rs
Comment (supernovahs) [inline crates/eigen-cli/src/generate.rs:95]: same

File: crates/eigen-cli/src/lib.rs
Comment (supernovahs) [inline crates/eigen-cli/src/lib.rs:187]: i think a better name would be convert_ecdsa to remove ambiguity

**Review by supernovahs — CHANGES_REQUESTED** [review]
some small nits

File: crates/eigen-cli/src/generate.rs
Comment (TomasArrachea) [reply crates/eigen-cli/src/generate.rs:15]: We took this constant from the Go implementation, and since this keys are for testing purposes we thought it's fine. Should we extend it?

File: crates/eigen-cli/src/lib.rs
Comment (TomasArrachea) [reply crates/eigen-cli/src/lib.rs]: Done!

File: crates/eigen-cli/src/generate.rs
Comment (dralves) [reply crates/eigen-cli/src/generate.rs:15]: ah, k. let's keep it the same then. thanks!

**Comment by supernovahs** [comment]
also @dralves  do we want the test naming convention to be test_name or just name? 

because the tests above  don't use test as prefix , while we are doing that in other crates .

**Comment by dralves** [comment]
> also @dralves do we want the test naming convention to be test_name or just name?
> 
> because the tests above don't use test as prefix , while we are doing that in other crates .

good call. let's keep it consistent

File: crates/eigen-cli/src/lib.rs
Comment (TomasArrachea) [reply crates/eigen-cli/src/lib.rs:85]: This prints are for showing the command's results, not for debugging. The same is done in the Go implementation. Should we change it?

**Comment by ricomateo** [comment]
@supernovahs @dralves I just fixed the test naming. Should we create a new PR fixing this issue in the other crates?

File: crates/eigen-cli/src/lib.rs
Comment (supernovahs) [reply crates/eigen-cli/src/lib.rs:85]: oh, ok ,can u add  a cfg(not test ) flag then, so it doesn't print on tests

Comment (supernovahs) [reply crates/eigen-cli/src/lib.rs:85]: all good

File: crates/eigen-cli/src/args.rs
Comment (supernovahs) [inline crates/eigen-cli/src/args.rs:58]:
```suggestion
all the relevant files for reading the key and decrypts it and also gets 
```

**Review by supernovahs — COMMENTED** [review]
^

File: crates/eigen-cli/src/args.rs
Comment (ricomateo) [reply crates/eigen-cli/src/args.rs]: Done!

**Review by supernovahs — APPROVED** [review]

**Comment by supernovahs** [comment]
Blocked by #69

**Comment by supernovahs** [comment]
@ricomateo  this is now unblocked, has some conflicts from bls crate sorry

**Pull request merged**
